fix(Designer): Floating Link Doesn't Disappear On Outer Click

Elaina-Lee opened the pull request:
- **Please check if the PR fulfills these requirements**

* [X] The commit message follows our guidelines

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

- **What is the current behavior?** (You can also link to an open issue here)
The floating link doesn't disappear
- **What is the new behavior (if this is a feature change)?**
The floating link disppaers on outer click except the editors
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

- **Please Include Screenshots or Videos of the intended change**:



https://github.com/Azure/LogicAppsUX/assets/144840522/24789bcf-3ccb-4f60-862f-fbbcec734255

Eric-B-Wu reviewed (changes requested):
I think this runs into an issue where once you create a link you cant actually edit that link again. I'll do some testing on your branch to confirm

Eric-B-Wu commented:
> I think this runs into an issue where once you create a link you cant actually edit that link again. I'll do some testing on your branch to confirm

yeah this is what im seeing on my end:
![linknolongershowing](https://github.com/Azure/LogicAppsUX/assets/95886809/e75ad74e-d7be-4061-9b69-e42806594dd3)

Elaina-Lee commented:
> > I think this runs into an issue where once you create a link you cant actually edit that link again. I'll do some testing on your branch to confirm
> 
> yeah this is what im seeing on my end: ![linknolongershowing](https://private-user-images.githubusercontent.com/95886809/301988906-e75ad74e-d7be-4061-9b69-e42806594dd3.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NjA1MzIsIm5iZiI6MTcwNzg2MDIzMiwicGF0aCI6Ii85NTg4NjgwOS8zMDE5ODg5MDYtZTc1YWQ3NGUtZDdiZS00MDYxLTliNjktZTQyODA2NTk0ZGQzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDIxMzcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3ZDFjM2Q4ZGNmNTczOTM0ZjI3N2NiMjU2NDE0ZTJjNmU3Yjk3ZWJiMDRiNzhmMmEwYzA2ODFkNjVjNTMyM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jyc8OIstOV1mYr3LguUuZq3nCjeHLmiuUDJ4DxNGbkQ) [ ![linknolongershowing](https://private-user-images.githubusercontent.com/95886809/301988906-e75ad74e-d7be-4061-9b69-e42806594dd3.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NjA1MzIsIm5iZiI6MTcwNzg2MDIzMiwicGF0aCI6Ii85NTg4NjgwOS8zMDE5ODg5MDYtZTc1YWQ3NGUtZDdiZS00MDYxLTliNjktZTQyODA2NTk0ZGQzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDIxMzcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3ZDFjM2Q4ZGNmNTczOTM0ZjI3N2NiMjU2NDE0ZTJjNmU3Yjk3ZWJiMDRiNzhmMmEwYzA2ODFkNjVjNTMyM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jyc8OIstOV1mYr3LguUuZq3nCjeHLmiuUDJ4DxNGbkQ) ](https://private-user-images.githubusercontent.com/95886809/301988906-e75ad74e-d7be-4061-9b69-e42806594dd3.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NjA1MzIsIm5iZiI6MTcwNzg2MDIzMiwicGF0aCI6Ii85NTg4NjgwOS8zMDE5ODg5MDYtZTc1YWQ3NGUtZDdiZS00MDYxLTliNjktZTQyODA2NTk0ZGQzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDIxMzcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3ZDFjM2Q4ZGNmNTczOTM0ZjI3N2NiMjU2NDE0ZTJjNmU3Yjk3ZWJiMDRiNzhmMmEwYzA2ODFkNjVjNTMyM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jyc8OIstOV1mYr3LguUuZq3nCjeHLmiuUDJ4DxNGbkQ) [ ](https://private-user-images.githubusercontent.com/95886809/301988906-e75ad74e-d7be-4061-9b69-e42806594dd3.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NjA1MzIsIm5iZiI6MTcwNzg2MDIzMiwicGF0aCI6Ii85NTg4NjgwOS8zMDE5ODg5MDYtZTc1YWQ3NGUtZDdiZS00MDYxLTliNjktZTQyODA2NTk0ZGQzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDIxMzcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3ZDFjM2Q4ZGNmNTczOTM0ZjI3N2NiMjU2NDE0ZTJjNmU3Yjk3ZWJiMDRiNzhmMmEwYzA2ODFkNjVjNTMyM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jyc8OIstOV1mYr3LguUuZq3nCjeHLmiuUDJ4DxNGbkQ)

Thank you, addressed this and made changes. Updated the video to reflect the part mentioned after fix.
